feature_engine/sanity_check/similar_columns.py

```diff
@@ -0,0 +1,162 @@
+from typing import Any, List
+import logging
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _is_dataframe,
+)
+
+
+class SimilarColumns(BaseEstimator, TransformerMixin):
+    """Ensure that similar columns are in test and train dataset.
+
+
+    Parameters
+    ----
+    None
+
+    """
+
+    def __init__(
+        self,
+        impute_with: Any = np.nan,
+        missing_values: str = "ignore",
+        drop_if_more_columns: bool = True,
+        add_if_less_columns: bool = True,
+        verbose: bool = False
+    ):
+
+        if missing_values not in ["raise", "ignore"]:
+            raise ValueError("missing_values takes only values 'raise' or 'ignore'.")
+
+        if not isinstance(drop_if_more_columns, bool):
+            raise ValueError("drop_if_more_columns takes only booleans True and False")
+
+        if not isinstance(add_if_less_columns, bool):
+            raise ValueError("add_if_less_columns takes only booleans True and False")
+
+        if not isinstance(verbose, bool):
+            raise ValueError("verbose takes only booleans True and False")
+
+        self.impute_with = impute_with
+        self.missing_values = missing_values
+        self.drop_if_more_columns = drop_if_more_columns
+        self.add_if_less_columns = add_if_less_columns
+        self.verbose = verbose
+
+    def _check_input(self, X: pd.DataFrame):
+        X = _is_dataframe(X)
+
+        if self.missing_values == "raise":
+            # check if dataset contains na
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        return X
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        """Fit columns schema
+
+        Parameters
+        ----------
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe
+
+        y: None
+            y is not needed for this transformer. You can pass y or None.
+
+        """
+        X = self._check_input(X)
+
+        self.variables_ = X.columns
```

Comment:
the fit method needs to learn a parameter called n_features_in_ with the number of columns in the train set. This is for convention with sklearn, and to pass the sklearn check_estimator checks. 

In this particular transformer, this parameter is a bit useless, but probably the checks will not pass if we do not add it.

Reply:
Ok for n_features_in, but to my understanding, our transformer will never pass sklearn check_estimator because it cannot apply to numpy array (only pandas dataframe, because it need columns name)

Reply:
> 
> 
> Ok for n_features_in, but to my understanding, our transformer will never pass sklearn check_estimator because it cannot apply to numpy array (only pandas dataframe, because it need columns name) .
> 
> All checks passed.

There is a workaround in the function is_dataframe to allow numpy arrays solely to be able to pass the check_estimator tests. See [this](https://github.com/solegalli/feature_engine/blob/master/feature_engine/dataframe_checks.py#L36).

And, we need to add a check_estimator test for this class as well. See [this as an example](https://github.com/solegalli/feature_engine/blob/master/tests/test_transformation/test_check_estimator_transformers.py).